### PR TITLE
Fix/simplify sync logic

### PR DESCRIPTION
**Pull request by joewalsh**

### CODE REVIEW

File: Wikipedia/Code/WMFSettingsMenuItem.m [hunk 1]
```diff
@@ -136,6 +136,16 @@ + (WMFSettingsMenuItem *)itemForType:(WMFSettingsMenuItemType)type {
                                        disclosureText:nil
                                            isSwitchOn:[SessionSingleton sharedInstance].dataStore.readingListsController.isSyncEnabled];
         }
+        case WMFSettingsMenuItemType_StorageAndSyncingDebug: {
+            return
+            [[WMFSettingsMenuItem alloc] initWithType:type
+                                                title:@"Reading list danger zone"
```
Comment (montehurd) [inline Wikipedia/Code/WMFSettingsMenuItem.m]: Looks like we need a localized string here.

Comment (nambatee) [reply Wikipedia/Code/WMFSettingsMenuItem.m]: This is only for debugging though

Comment (montehurd) [reply Wikipedia/Code/WMFSettingsMenuItem.m]: Haha you're right! I just saw a string and didn't even read it :-P

Comment (joewalsh) [reply Wikipedia/Code/WMFSettingsMenuItem.m]: simulator screen shot - iphone x - 2018-02-21 at 07 45 52

**Pull request merged**